torch/csrc/distributed/autograd/context/dist_autograd_container.h

```diff
@@ -36,8 +36,15 @@ class TORCH_API DistAutogradContainer {
   const DistAutogradContext& newContext();
 
   // Clean up resources for a given context_id once the autograd pass is done.
+  // Sends RPC to other workers this worker knows about, telling them to clean
+  // up their context as well.
   void releaseContext(int64_t context_id);
 
+  // Releases an autograd context if it is present on this node. Also sends RPC
+  // to other workers this worker knows about, telling them to clean up their
+  // context. Does nothing if it is not present.
+  void releaseContextIfPresent(int64_t context_id);
+
   // Retrieve the autograd context for a given context_id.
```

Comment:
wondering why they can not be merged?

Reply:
One checks if the context exists, one throws if it does not. We can merge them if we have a flag `throwIfNotExists` or something similar, should we do this? cc @pritamdamania87

Reply:
Two separate methods seems a bit cleaner to me than having a flag.